[Mobile Payments] - Order Created Notice displayed after dismissing error modal

toupper opened the pull request:
<!-- Remember about a good descriptive title. -->

ℹ️ Feel free to merge it when reviewed as I will be AFK.

Closes: #8434 
<!-- Id number of the GitHub issue this PR addresses. -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
When trying to collect a payment whose amount was not valid (e.g it was less than the minimum amount allowed) we called to finish the whole collect payment flow (not only the card flow) process as if it was completed in the `CollectOrderPaymentUseCase`. We then interpreted it as if the payment was completed and show the Order Completed notice in `SimplePaymentsAmountHostingController`, leaving the users confused.
With this PR we don't call the completion closure, but the cancellation one. This doesn't only fix the issue described above but leaves the user in the payment methods screen to retry the payment with a different payment method. This is a more optimal flow than canceling the whole payment collection.

## Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

1. Go to payments and create a simple payment with the amount $0.01
2. Use Card as payment method
3. Attempt to complete payment - error about minimum amount displayed
4. Dismiss the error
5. User returns to IPP screen but the Order Completed toast is still displayed

## Videos
<!-- Include before and after images or gifs when appropriate. -->
### Before
https://user-images.githubusercontent.com/17252150/208022216-57a716bd-61fa-489f-a6ee-14afed166bbb.MP4

### After

https://user-images.githubusercontent.com/1864060/208136625-d8e247c9-8a8f-41eb-aa43-897162b87b48.MP4

---
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

toupper committed:
Do not complete the flow if the amount is not valid. Just cancel.

iamgabrielma committed:
Merge branch 'trunk' into issue/8434-Order-Completed-Notice-When-Error

iamgabrielma commented:
Works great! I'll hold the merge for the moment as we're having some tests failing in trunk. Ref: p1671436724169599-slack-C02KUCFCSFP/

iamgabrielma committed:
Merge branch 'trunk' into issue/8434-Order-Completed-Notice-When-Error

iamgabrielma committed:
onCancel promise for below minimum collectPayment

As we call the `onCancel` completion handler now rather than `onCompleted when collectPayment is below minimum amount, this test will fail as the promise() call returns nil when unwrapping the optional.

iamgabrielma reviewed (approved):
Merged trunk into this branch to update failing unit tests, and fixed additional test on 359afd389494e6dd51cb7fbfc771ccafd8bc6b55
